internal/telemetry/tracker.go

```diff
@@ -0,0 +1,56 @@
+package telemetry
+
+import (
+	"fmt"
+)
+
+// Tracker logs events
+type Tracker interface {
+	Track(event Event)
+}
+
+type tracker struct {
+	userID          string
+	executionID     string
+	telemetryClient telemetryClient
```

Comment:
this all looks fine, implementation wise.  I'd just like to reorganize and rename here, and I think we can combine the logic from `NewEvent` into what's going on here (and completely omit any event constructors being exported from `telemetry`:

```go
type Service interface {
  TrackEvent(eventType EventType, data ...EventData)
}

type service struct {
  command string
  executionID string
  userID string
  tracker Tracker
}
```

Reply:
Ok, but doing that kind of made the event_test.go useless, so I removed that file.  Should I try and create a tracker_test.go instead now?  Should I add a testTracker struct to help with testing purposes?

Reply:
yeah that all sounds perfect... i'll take another look once you get that new test up

Reply:
@makesitgo ok just added the tests